code/game/objects/items/weapons/legcuffs.dm

```diff
@@ -94,42 +99,58 @@
 	return TRUE
 
 /obj/item/restraints/legcuffs/beartrap/Crossed(AM as mob|obj, oldloc)
-	if(armed && isturf(src.loc))
-		if( (iscarbon(AM) || isanimal(AM)) && !istype(AM, /mob/living/simple_animal/parrot) && !isconstruct(AM) && !isshade(AM) && !istype(AM, /mob/living/simple_animal/hostile/viscerator))
-			var/mob/living/L = AM
-			armed = FALSE
-			update_icon()
-			playsound(src.loc, 'sound/effects/snap.ogg', 50, 1)
-			if(!silent_arming)
-				L.visible_message("<span class='danger'>[L] triggers \the [src].</span>", \
-						"<span class='userdanger'>You trigger \the [src]!</span>")
-
-			if(IED && isturf(src.loc))
-				IED.active = TRUE
-				message_admins("[key_name_admin(usr)] has triggered an IED-rigged [name].")
-				log_game("[key_name(usr)] has triggered an IED-rigged [name].")
-				spawn(IED.det_time)
-					IED.prime()
-
-			if(sig && isturf(src.loc))
-				sig.signal()
-
-			if(ishuman(AM))
-				var/mob/living/carbon/H = AM
-				if(IS_HORIZONTAL(H))
-					H.apply_damage(trap_damage, BRUTE,"chest")
-				else
-					H.apply_damage(trap_damage, BRUTE,(pick("l_leg", "r_leg")))
-				if(!H.legcuffed && H.get_num_legs() >= 2) //beartrap can't cuff you leg if there's already a beartrap or legcuffs.
-					H.legcuffed = src
-					forceMove(H)
-					H.update_inv_legcuffed()
-					SSblackbox.record_feedback("tally", "handcuffs", 1, type)
+	if(!armed || !isturf(loc))
+		return ..()
+
+	if((iscarbon(AM) || isanimal(AM)) && !istype(AM, /mob/living/simple_animal/parrot) && !isconstruct(AM) && !isshade(AM) && !istype(AM, /mob/living/simple_animal/hostile/viscerator))
```

Comment:
```suggestion
	if((iscarbon(AM) || isanimal(AM)) && AM.flying != TRUE))
```
Wouldn't this be exactly what Henri asked?

Reply:
Adjusted in [81a2414](https://github.com/ParadiseSS13/Paradise/pull/21941/commits/81a241446d343b2d0b9bae1efa0b926e51dc2fa5)